cpp/src/IceGrid/Activator.cpp

```diff
@@ -641,20 +641,50 @@ Activator::activate(const string& name,
     }
 
     vector<gid_t> groups;
+#ifdef _AIX
+    char* grouplist = getgrset(pw->pw_name);
+    if(grouplist == 0)
+    {
+        throw SyscallException(__FILE__, __LINE__, getSystemErrno());
+    }
+    vector<string> grps;
+    if(IceUtilInternal::splitString(grouplist, ",", grps))
+    {
+        for(vector<string>::const_iterator p = grps.begin(); p != grps.end(); ++p)
+        {
+            gid_t group;
+            istringstream is(*p);
+            if(is >> group)
+            {
+                groups.push_back(group);
+            }
+        }
+    }
+#else
     groups.resize(20);
     int ngroups = static_cast<int>(groups.size());
-#if defined(__APPLE__)
+    #if defined(__APPLE__)
     if(getgrouplist(pw->pw_name, static_cast<int>(gid), reinterpret_cast<int*>(&groups[0]), &ngroups) < 0)
-#else
+    #else
     if(getgrouplist(pw->pw_name, gid, &groups[0], &ngroups) < 0)
-#endif
+    #endif
     {
         groups.resize(static_cast<size_t>(ngroups));
-#if defined(__APPLE__)
+        #if defined(__APPLE__)
         getgrouplist(pw->pw_name, static_cast<int>(gid), reinterpret_cast<int*>(&groups[0]), &ngroups);
-#else
+        #else
         getgrouplist(pw->pw_name, gid, &groups[0], &ngroups);
+        #endif
+    }
+    else
+    {
+       groups.resize(static_cast<size_t>(ngroups));
+    }
 #endif
+
+    if(groups.size() > NGROUPS_MAX)
+    {
+        groups.resize(NGROUPS_MAX);
     }
     else
```

Comment:
How this works, seems `ngroups` is not defined for AIX

Reply:
Looks like it was a merge bug, Bernard fixed it.